platforms/documentation/docs/src/docs/userguide/best-practices/bp_tasks.adoc

```diff
@@ -121,3 +121,66 @@ include::sample[dir="snippets/bestPractices/useCacheabilityAnnotations/groovy",f
 
 === Tags
 `#tasks`, `#caching`
+
+== Don't resolve Configurations before Task Execution
+
+Resolving configuration before task execution phase can lead to incorrect results and slows gradle build performance.
+
+Once configuration is resolved it can't be resolved again, and it can lose some information when it is resolved too early.
```

Comment:
I think this needs more detail.

Reply:
When I wrote this, I thought about two potential problems:
* task dependencies loss
* items added/modified to/in configuration after resolution

But the second part is very-well guarded by Gradle's runtime. it reports errors and warnings when something fishy is happening with Configuration.  So perhaps I can cut this part and focus only on the task dependencies, as this info is not tracked by Gradle runtime (yet).